### ISSUE TITLE
Could relays resolve names when adding faces?

### ISSUE THREAD

**Issue by Wojtek242** [cn-uofbasel/ccn-lite#219]
I am trying to get ccn-lite working in docker containers and I am using docker-compose to network them at the moment. One convenient thing about docker-compose is that it allows the containers to access each other via their container names instead of an IP address. The way it works is that DNS requests for names that match container names will get correctly resolved by the network setup by docker-compose. 

I noticed that this name resolution did not happen when adding a face to a relay. Would it make sense to add this capability?

This way I could set up a docker-compose file with nodeA, nodeB, nodeC, ... and add faces using their names rather than IP addresses.

**Comment by mfrey** [comment]
> I noticed that this name resolution did not happen when adding a face to a relay. Would it make sense to add this capability?

Short answer: It might make sense, depending on the use case (e.g. it eases your docker setup) and the implications for the face generation process.

Slightly longer answer: A face is not necessarily a network interface. It could be  a IPC channel to an application (pipes, shared memory, etc.) , it could be an interface directly to hardware (IEEE 802.15.4 radio, bluetooth adapter, etc.) or an inter-node channel (such as tunnels, etc.) or even something completely else.

**Comment by blacksheeep** [comment]
But I think for udp sockets that should be possible?

**Comment by mfrey** [comment]
> But I think for udp sockets that should be possible?

Absolutely, but do you really need it (except for something like docker)?

**Comment by blacksheeep** [comment]
I think for IP sockets, it should be possible to add a DNS lookup without affecting the others or changing a lot of code. So if somebody wants to write a patch for that... :-)

**Comment by Wojtek242** [comment]
I guess this and the previous issue I raised could both be solved by using UDP sockets, but admittedly it does confine you to the IP/UDP stack. Perhaps this could be an option, but I understand the reasons now
